geoportal/geoportailv3_geoportal/lib/lingua_extractor.py

```diff
@@ -0,0 +1,378 @@
+            'FEATURE_COUNT': '50',
+            'X': x,
+            'Y': y,
+            'SRS': srs,
+            'WIDTH': width,
+            'HEIGHT': height,
+            'BBOX': bbox
+        }
+        metadata = session.query(Metadata).filter(Metadata.item_id == layer_id).\
+                   filter(Metadata.name == "ogc_layers").first()
+        if metadata is not None:
+            body['LAYERS'] = metadata.value
+
+        metadata = session.query(Metadata).filter(Metadata.item_id == layer_id).\
+                   filter(Metadata.name == "ogc_query_layers").first()
+        if metadata is not None:
+            body['QUERY_LAYERS'] = metadata.value
+
+        metadata = session.query(Metadata).filter(Metadata.item_id == layer_id).\
+                   filter(Metadata.name == "ogc_info_format").first()
+        if metadata is not None:
+            body['INFO_FORMAT'] = metadata.value
+        ogc_info_srs = "epsg:2169"
+        metadata = session.query(Metadata).filter(Metadata.item_id == layer_id).\
+                   filter(Metadata.name == "ogc_info_srs").first()
+        if metadata is not None:
+            ogc_info_srs = metadata.value
+
+        separator = "?"
+        if url.find(separator) > 0:
+            separator = "&"
+        query = '%s%s%s' % (url, separator, urlencode(body))
+        content = ""
+        try:
+            result = urllib.request.urlopen(query, None, 15)
+            content = result.read()
+        except Exception as e:
+            log.exception(e)
+            return []
+        try:
+            ogc_features = geojson_loads(content)
+            return (dict((key, key)
+                         for key, value in ogc_features['features'][0]['properties'].items()))
+        except Exception as e:
+            log.exception(e)
+            return []
+        return []
+
+    def _extract_messages(self, session):  # pragma: nocover
+        print('Entering into the tooltip extractor')
+
+        from c2cgeoportal_commons.models import DBSession, DBSessions
+        from geoportailv3_geoportal.models import LuxGetfeatureDefinition, LuxLayerInternalWMS
+
+        results = session.query(LuxGetfeatureDefinition).\
+                  filter(LuxGetfeatureDefinition.remote_template == False).\
+                  filter(
+                      LuxGetfeatureDefinition.template.in_
+                      (['default.html', 'default_table.html', 'feedbackanf.html']))  # noqa
+
+        print("%d results" % results.count())
+        for result in results:
+            engine = DBSessions[result.engine_gfi]
+            first_row = None
+            if result.query is not None and len(result.query) > 0:
+                try:
+                    if "SELECT" in result.query.upper():
+                        first_row = engine.execute(result.query).first()
+                    else:
+                        first_row = engine.execute("SELECT * FROM " + result.query).first()
```

Comment:
This my friend, is SQL injection, via SQL :exploding_head: 

When using raw SQL, I tend to wrap everything with the `text()` function of SQLAlchemy.

Reply:
this is a legacy configuration, the configuration DB can contain raw SQL queries. (it is controlled by an admin though)
Hard to refactor
we should discuss this

Reply:
At least using the `text()` function would mitigate a lot of cases.